src/embind/emval.js

```diff
@@ -174,7 +174,7 @@ var LibraryEmVal = {
 
     for(var i = 0; i < argCount; ++i) {
         functionBody +=
-            "var argType"+i+" = requireRegisteredType(HEAP32[(argTypes >> 2) + "+i+"], \"parameter "+i+"\");\n" +
+            "var argType"+i+" = requireRegisteredType(Module['HEAP32'][(argTypes >> 2) + "+i+"], \"parameter "+i+"\");\n" +
```

Comment:
I don't think HEAP32 is closed over, unless I'm missing something? It should be using the HEAP32 from the outermost scope, which means the proper one?

Do you have a testcase showing the issue?

Reply:
To clarify, `functionBody` returns another function that closes over the `HEAP32` argument passed to `functionBody`, and the returned function is what triggers the exception.

Reply:
Ok, now that I understand the issue, I see what you meant here. However, I think it would be better to not close over HEAP32 entirely - we should just not pass it as a parameter, we can use it from the outer scope?

Reply:
Because the function is created with `new Function(...)`, it knows only about the global scope and not the enclosing one. That will work for the test case because `HEAP32` is in the global scope, but won't work if the build output is wrapped in another scope.
Using `eval()` instead would give it access to the enclosing scope.

Reply:
I see, thanks